### DISCUSSION TITLE
How best to use pi4j-example-components?

### DISCUSSION THREAD

**Discussion by fovea1959** [Pi4J/pi4j#275] [Q&A]
I have an existing gradle project, and need to use pi4j in it; specifically, I need to use the ads1115.

pi4j-example-components has the code to read the ads1115, but what is the best way to bring pi4j-example-components into an existing project? Just copy the source code in? Include it as a git subproject? something else? I don't see a maven artifact for it...

**Comment by FDelporte** [comment]
pi4j-example-components isn't a maven artifact indeed, but a collection of examples to copy-paste-adapt.

https://pi4j.com/getting-started/minimal-example-application/ links to sources of a project that contains both Maven and Gradle files to show how to add the Pi4J dependencies.

I hope this gets you started...
